### PR TITLE
init dataconnect:sdk and other SDK onboarding improvements

### PR DESCRIPTION
**Pull request by joehan**
### Description
- Adds `init dataconnect:sdk` to help guide users on how to configure their generated SDKs.
- Calls out this new command at the end of `init dataconnect`
- Adds a commented out example of a 'generate' block to the template `connector.yaml`

### Scenarios Tested
I tested out adding SDKs of each type. I also confirmed that if you run this with existing SDKs configured, we default to the values you already have configured.

### CODE REVIEW

File: src/commands/dataconnect-sdk-generate.ts [hunk 1]
```diff
@@ -31,7 +32,10 @@ export const command = new Command("dataconnect:sdk:generate")
       if (!hasGeneratables) {
         logger.warn("No generated SDKs have been declared in connector.yaml files.");
         logger.warn(
-          "See https://firebase.google.com/docs/data-connect/quickstart#configure-sdk-outputs for examples of how to configure generated SDKs.",
+          `Run ${clc.bold("firebase init dataconnect:sdk")} to configure a generated SDK.`,
+        );
+        logger.warn(
+          `See https://firebase.google.com/docs/data-connect/quickstart#configure-sdk-outputs for more details of how to configure generated SDKs.`,
```
Comment (fredzqm) [inline src/commands/dataconnect-sdk-generate.ts:38]: @markarndt FYI on the link.

It doesn't work right now. If you have a "future" link, that would be neat.

File: src/init/features/dataconnect/index.ts [hunk 2]
```diff
@@ -96,6 +99,10 @@ export async function doSetup(setup: Setup, config: Config): Promise<void> {
       waitForCreation: false,
     });
   }
+  logger.info("");
```
Comment (fredzqm) [inline src/init/features/dataconnect/index.ts:102]: Did you forget to remove this?

File: templates/_gitignore [hunk 3]
```diff
@@ -64,3 +64,6 @@ node_modules/
 
 # dotenv environment variables file
 .env
+
+# dataconnect generated files
+.dataconnect
```
Comment (fredzqm) [inline templates/_gitignore:69]: nit: empty line at the end

File: src/commands/init.ts [hunk 4]
```diff
@@ -85,6 +85,11 @@ choices.push({
   name: "Data Connect: Set up a Firebase Data Connect service.",
   checked: false,
 });
+choices.push({
+  value: "dataconnect:sdk",
+  name: "Data Connect: Set up a generated SDK for your Firebase Data Connect service.",
```
Comment (fredzqm) [inline src/commands/init.ts:90]: Data Connect SDK: Set up a generated SDK for your Firebase Data Connect service.

Screenshot 2024-06-25 at 5 28 38 PM

**Comment by fredzqm** [comment]
Did a test run. Looks amazing!

Screenshot 2024-06-25 at 5 33 44 PM

**Review by fredzqm — APPROVED** [review]

File: src/init/features/dataconnect/index.ts (hunk 2 above)
Comment (joehan) [reply src/init/features/dataconnect/index.ts:102]: Nope, added this to space out the next message a little better so it is more visible

File: src/commands/init.ts (hunk 4 above)
Comment (joehan) [reply src/commands/init.ts:90]: Followed the pattern used by hosting here, which has 2 init options:
Screenshot 2024-06-25 at 6 05 22 PM

File: src/commands/dataconnect-sdk-generate.ts (hunk 1 above)
Comment (joehan) [reply src/commands/dataconnect-sdk-generate.ts:38]: Dang, this link previously worked, looks like we edited that page recently. Linking to https://firebase.google.com/docs/data-connect/gp/web-sdk instead for now, but feel free to edit suggest a better link @markarndt

File: templates/init/dataconnect/connector.yaml [hunk 5]
```diff
@@ -1,2 +1,13 @@
...
+## ## You'll need to replace the outputDirs with ones pointing to where you want the generated code in your app.
+# generate:
+#   javascriptSdk:
+#     outputDir: <Path where you want the generated SDK to be written to, relative to this file>
+#     package: "@firebasegen/my-connector"
+#     packageJSONDir: < Optional. Path to your Javascript app's package.json>
+#   swiftSdk:
+#     outputDir: <Path where you want the generated SDK to be written to, relative to this file>
+#   kotlinSdk:
+#     outputDir: <Path where you want the generated SDK to be written to, relative to this file>
```
Comment (dconeybe) [inline templates/init/dataconnect/connector.yaml:13]: nit: add "package" under "kotlinSdk" to specify the kotlin package to incorporate into the generated kotlin code.

**Pull request merged**